src/utils/version.ts

```diff
@@ -0,0 +1,52 @@
+import { isNumber } from 'lodash'
+
+const versionPattern = /^(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:-([0-9A-Za-z\.]+))?/
+
+export class SemVersion {
```

Comment:
Could we not just use the `semver` package for this?

Reply:
Iirc the `semver` package does not handle [pre-release versions well](https://github.com/npm/node-semver#prerelease-tags). We could double check but I think we've ran into these kinds of issues before. I'm not a big fan of the duplicated code so if we can get this working it'll be nicer.